src/ray/common/id.h

```diff
@@ -176,6 +177,9 @@ class TaskID : public BaseID<TaskID> {
   static constexpr size_t kUniqueBytesLength = 8;
 
  public:
+  static_assert(absl::little_endian::IsLittleEndian(),
```

Comment:
Question: This is a compilation check, so if the binary is compiled under little-endian byte order and run under big-endian byte order, this assertion won't fail. I think that this would be an issue for dynamically switching byte orders e.g. on ARM.

I _think_ that cross-compilation would be fine given that this static assertion consults the compiler for endianness, and this should be set to the target platform's endianness if the world is sensible?

Anyways, static check should be fine for us I think. 😄